app/src/main/java/com/github/se/travelpouch/ui/dashboard/AddActivityScreen.kt

```diff
@@ -0,0 +1,198 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.OffsetMapping
+import androidx.compose.ui.text.input.TransformedText
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.core.text.isDigitsOnly
+import com.github.se.travelpouch.model.Location
+import com.github.se.travelpouch.model.activity.Activity
+import com.github.se.travelpouch.model.activity.ActivityModelView
+import com.google.firebase.Timestamp
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.GregorianCalendar
+import java.util.Locale
+
+
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+@Composable
+fun AddActivity(activityModelView: ActivityModelView){
+
+    var title by remember { mutableStateOf("") }
+    var description by remember { mutableStateOf("") }
+    var dateText by remember { mutableStateOf("") }
+    var location by remember { mutableStateOf("Placeholder") }
+
+    val context = LocalContext.current
+
+    val placeholerLocation = Location(
+        0.0,
+        0.0,
+        Timestamp(0, 0),
+        "name"
+    )
+
+    Scaffold {
+        Column(
+            modifier = Modifier.fillMaxSize().padding(16.dp).verticalScroll(rememberScrollState()),
+            verticalArrangement = Arrangement.spacedBy(16.dp)
+        ) {
+            OutlinedTextField(
+                value = title,
+                onValueChange = { title = it },
+                label = { Text("Title") },
+                placeholder = { Text("Title") }
+            )
+
+            OutlinedTextField(
+                value = description,
+                onValueChange = { description = it },
+                label = { Text("Description") },
+                placeholder = { Text("Description") }
+            )
+
+            OutlinedTextField(
+                value = dateText,
+                onValueChange = {
+                    if(it.isDigitsOnly() && it.length <= 8){
+                        dateText = it
+                    }
+                },
+                label = { Text("Date") },
+                placeholder = { Text("01/01/1970") },
+                visualTransformation = DateVisualTransformation(),
+                keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number)
+            )
+
+            OutlinedTextField(
+                value = location,
+                onValueChange = {  },
+                label = { Text("Location") },
+                placeholder = { Text("Location") }
+            )
+
+            Button(
+                enabled = location.isNotBlank() && title.isNotBlank() && description.isNotBlank()
+                        && dateText.isNotBlank(),
+                onClick = {
+                    try{
```

Comment:
It would be better to consider adding more specific error handling for the date parsing and the Firebase operation failure

Reply:
Can you clarify this point ? Because this operation is not related to Firebase. Before interacting with the modelView, we are tryign that the date on the text field is valid or not. For example it verifies (before calling addActivity) that the date is not 00/00/0000. so I don't see what a failure Firebase operation would do here

Reply:
Ok, no I understand. Well, we can pass a context parameter to the modelView to directly show a toast from the modelView (but displayed on the addActivity screen). And, we can separate the exception of the parsing from the failure of the model view. Good catch

Reply:
My suggestion was more about distinguishing parsing errors from activity creation errors and Firestore operations. For example, if Firestore fails, the user should receive a toast saying 'Unable to add the activity' instead of a message about an invalid date